### DISCUSSION TITLE
equinox + equinox.binaries

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/equinox#219] [General]
Currently one has to checkout both repos and give a reference where the side-by-side checkout is located, see for example:
- https://github.com/eclipse-equinox/equinox/pull/212

I wonder if instead we should use [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) instead, (e.g. m2e uses that to pull in the tests-repository) to makes this explicit?

**Comment by akurtakov** [comment]
That is probably good idea. The only complication might be to test to fix https://github.com/eclipse-platform/eclipse.platform.releng.aggregator to work with submodule in submodule.
IMHO even better solution would be to get rid of the binaries repo via an approach like https://github.com/eclipse-platform/eclipse.platform.swt.binaries/issues/2#issuecomment-1085567970 (which I never found time to work on). After all Git is not the best way to store dll/so files.

**Reply by laeubi** [reply]
Technically we might want to deploy those files to the eclipse-repo (e.g. in the form of a zip file), then it could be downloaded by the build and expanded, advantage would be we do not need another communication channel in the build.
